src/main/webapp/cluster-view/cluster-view.controller.js

```diff
@@ -1,26 +1,150 @@
 'use strict';
 
 angular.module('managementConsole')
-    .controller('ClusterViewCtrl', [
+  .controller('ClusterViewCtrl', [
     '$scope',
     '$stateParams',
     '$state',
+    '$q',
     'modelController',
-    function ($scope, $stateParams, $state, modelController) {
-            if (!modelController.isAuthenticated()) {
-                $state.go('/logout');
+    function ($scope, $stateParams, $state, $q, modelController) {
+      if (!modelController.isAuthenticated()) {
+        $state.go('/logout');
+      }
+      $scope.shared = {
+        currentCollection: 'caches'
+      };
+      $scope.clusters = modelController.getServer().getClusters();
+      $scope.currentCluster = modelController.getServer().getCluster($scope.clusters, $stateParams.clusterName);
+
+      $scope.$watch('currentCluster', function (currentCluster) {
+        if (currentCluster && currentCluster.name !== $stateParams.clusterName) {
+          $state.go('clusterView', {
+            'clusterName': currentCluster.name
+          });
+        }
+      });
+
+      $scope.currentClusterState = function () {
+        return $scope.currentClusterAvailability() ? 'AVAILABLE' : 'UNAVAILABLE';
+      };
+
+      $scope.currentClusterAvailability = function () {
+        return ispn.util.isNotNullOrUndefined($scope.currentCluster) && $scope.currentCluster.availability === 'AVAILABLE';
+      };
+
+      $scope.currentClusterHasNodes = function () {
+        return ispn.util.isNonEmptyArray($scope.currentCluster.getNodes());
+      };
+
+      $scope.currentClusterHasCaches = function () {
+        return !ispn.util.isEmptyObject($scope.currentCluster.getCaches());
+      };
+
+      $scope.isNodesView = function () {
+        return $scope.shared.currentCollection === 'nodes';
+      };
+
+      $scope.isCachesView = function () {
+        return $scope.shared.currentCollection === 'caches';
+      };
+
+      $scope.zoomValue = 0;
+
+      $scope.isZoomedIn = function () {
+        return $scope.zoomValue == 0 ? false : true;
+      };
+
+      $scope.initialMin = 0;
+      $scope.initialMax = 1000;
+      $scope.sliders = [
+        {
+          name: "Average read time",
+          id: 'average-read-time',
+          from: $scope.initialMin,
+          to: $scope.initialMax
+        },
+        {
+          name: "Average write time",
+          id: 'average-write-time',
+          from: $scope.initialMin,
+          to: $scope.initialMax
+        },
+        {
+          name: "Average remove time",
+          id: 'average-remove-time',
+          from: $scope.initialMin,
+          to: $scope.initialMax
+        },
+        {
+          name: "Average replication time",
+          id: 'average-replication-time',
+          from: $scope.initialMin,
+          to: $scope.initialMax
+        }
+      ];
+
+      angular.forEach($scope.sliders, function (slider) {
```

Comment:
While this is ok, even more cool approach would be to implement `onChange` attribute for slider that would execute given expression when slider value changes, very similar to what `ng-change` does, so you would not have to do watch here.